tests/app/helpers/test_template_helpers.py

```diff
@@ -64,114 +161,299 @@ def test_get_footer_context_nisra_theme(app):
+            include_csrf_token=True,
+            survey_config=survey_config,
+        ).context["account_service_url"]
+
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "theme,language,expected",
+    [
+        ("default", "en", SurveyConfig),
+        ("default", "cy", SurveyConfig),
+        ("business", "en", SurveyConfig),
+        ("business", "cy", SurveyConfig),
+        ("health", "en", SurveyConfig),
+        ("health", "cy", SurveyConfig),
+        ("social", "en", SurveyConfig),
+        ("social", "cy", SurveyConfig),
+        ("northernireland", "en", SurveyConfig),
+        ("northernireland", "cy", SurveyConfig),
+        ("census", "en", CensusSurveyConfig),
+        ("census", "cy", WelshCensusSurveyConfig),
+        ("census-nisra", "en", CensusNISRASurveyConfig),
+        ("census-nisra", "cy", CensusNISRASurveyConfig),
+    ],
+)
+def test_get_survey_config(
+    app: Flask, theme: str, language: str, expected: SurveyConfig
+):
+    with app.app_context():
+        result = get_survey_config(theme=theme, language=language)
+    assert isinstance(result, expected)
 
-def test_get_census_base_url_welsh():
-    expected = "https://cyfrifiad.gov.uk/"
-    result = get_census_base_url(schema_theme="census", language_code="cy")
 
-    assert expected == result
+def test_context_set_from_app_config(app):
+    with app.app_context():
+        current_app.config["COOKIE_SETTINGS_URL"] = "test-cookie-settings-url"
+        current_app.config["CDN_URL"] = "test-cdn-url"
+        current_app.config["CDN_ASSETS_PATH"] = "/test-assets-path"
+        current_app.config["ADDRESS_LOOKUP_API_URL"] = "test-address-lookup-api-url"
+        current_app.config["EQ_GOOGLE_TAG_MANAGER_ID"] = "test-google-tag-manager-id"
+        current_app.config[
+            "EQ_GOOGLE_TAG_MANAGER_AUTH"
+        ] = "test-google-tag-manager-auth"
+        survey_config = SurveyConfig()
+
+        context = ContextHelper(
+            language="en",
+            is_post_submission=False,
+            include_csrf_token=True,
+            survey_config=survey_config,
+        ).context
+
+    assert context["cookie_settings_url"] == "test-cookie-settings-url"
+    assert context["cdn_url"] == "test-cdn-url/test-assets-path"
+    assert context["address_lookup_api_url"] == "test-address-lookup-api-url"
+    assert context["google_tag_manager_id"] == "test-google-tag-manager-id"
+    assert context["google_tag_manager_auth"] == "test-google-tag-manager-auth"
+
+
+@pytest.mark.parametrize(
+    "theme,language,expected",
+    [
+        ("default", "en", "main"),
+        ("business", "en", "main"),
+        ("health", "en", "main"),
+        ("social", "en", "main"),
+        ("northernireland", "en", "main"),
+        ("census", "en", "census"),
+        ("census", "cy", "census"),
+        ("census-nisra", "en", "census"),
+    ],
+)
+def test_correct_theme_in_config(app: Flask, theme: str, language: str, expected: str):
```

Comment:
This is checking the theme in the context rather than the config.
```suggestion
def test_correct_theme_in_context(app: Flask, theme: str, language: str, expected: str):
```